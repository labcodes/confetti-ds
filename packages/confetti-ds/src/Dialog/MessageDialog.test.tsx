/* eslint-disable no-alert */
import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { MessageDialog } from "./index";

const args = {
  title: "Sample Message Dialog",
  icon: "star" as const,
  content:
    "This is the main content of a Message Dialog. You may want to put as much content as you want, as long as it's a string.",
  buttonProps: {
    label: "Required button",
    onClick: () => alert("Required button clicked"),
  },
  outlineButtonProps: {
    label: "Optional button",
    onClick: () => alert("Optional button clicked"),
  },
  isModal: false,
  isOpen: false,
  handleCLose: jest.fn(),
  isMessageDialog: true,
};

describe("MessageDialog", () => {
  afterEach(() => {
    /* reset html tag after each test case */
    document.getElementsByTagName("html")[0].innerHTML = "";
  });

  it("rendering with base props", () => {
    expect(document.body.style.overflow).toBe("");

    const Component = <MessageDialog {...args} />;
    const renderedComponent = renderer.create(Component);

    expect(renderedComponent.toJSON()).toMatchSnapshot();

    /* To clean up things */
    renderedComponent.unmount();
  });

  it("rendering with isOpen false", () => {
    expect(document.body.style.overflow).toBe("");

    const Component = (
      <MessageDialog {...args} isModal isLarge isOpen={false} />
    );
    const renderedComponent = renderer.create(Component);

    expect(renderedComponent.toJSON()).toMatchSnapshot();

    /* To clean up things */
    renderedComponent.unmount();

    expect(document.body.style.overflow).toBe("");
  });

  it("rendering with isModal false", () => {
    expect(document.body.style.overflow).toBe("");

    const Component = (
      <MessageDialog {...args} isModal={false} isLarge isOpen />
    );
    const renderedComponent = renderer.create(Component);

    expect(renderedComponent.toJSON()).toMatchSnapshot();

    /* To clean up things */
    renderedComponent.unmount();

    expect(document.body.style.overflow).toBe("unset");
  });

  it("rendering with isLarge false", () => {
    expect(document.body.style.overflow).toBe("");

    const Component = (
      <MessageDialog {...args} isModal isOpen isLarge={false} />
    );
    const renderedComponent = renderer.create(Component);

    expect(renderedComponent.toJSON()).toMatchSnapshot();

    /* To clean up things */
    renderedComponent.unmount();

    expect(document.body.style.overflow).toBe("unset");
  });

  it(`rendering if isOpen false sets body overflow to empty`, () => {
    expect(document.body.style.overflow).toBe("");

    const wrapper = mount(
      <MessageDialog
        {...args}
        isModal // true
        isOpen={false}
      />
    );

    expect(document.body.style.overflow).toBe("");

    expect(wrapper.find(".lab-dialog__title").exists()).toBeFalsy();
    expect(wrapper.find(".lab-dialog__body").exists()).toBeFalsy();

    wrapper.unmount();
  });
  it("checks if isOpen and isModal true sets body overflow to hidden", () => {
    expect(document.body.style.overflow).toBe("");

    const wrapper = mount(
      <MessageDialog
        {...args}
        isModal // true
        isOpen // true
      />
    );

    expect(document.body.style.overflow).toBe("hidden");

    expect(wrapper.find(".lab-dialog__title").text()).toBe(args.title);
    expect(wrapper.find(".lab-dialog__body").text()).toBe(args.content);

    wrapper.unmount();

    expect(document.body.style.overflow).toBe("unset");
  });
  it("toggle isOpen", () => {
    expect(document.body.style.overflow).toBe("");

    const wrapper = mount(<MessageDialog {...args} isModal isOpen />);

    expect(document.body.style.overflow).toBe("hidden");

    expect(wrapper.find(".lab-dialog__title").text()).toBe(args.title);
    expect(wrapper.find(".lab-dialog__body").text()).toBe(args.content);

    wrapper.setProps({ isOpen: false });

    expect(document.body.style.overflow).toBe("unset");

    expect(wrapper.find(".lab-dialog__title").exists()).toBeFalsy();
    expect(wrapper.find(".lab-dialog__body").exists()).toBeFalsy();

    wrapper.unmount();

    expect(document.body.style.overflow).toBe("unset");
  });
});
