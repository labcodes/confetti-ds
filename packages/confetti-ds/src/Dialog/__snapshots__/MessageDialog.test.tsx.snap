// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageDialog rendering with base props 1`] = `null`;

exports[`MessageDialog rendering with isLarge false 1`] = `
[
  <div
    className="lab-dialog-overlay"
    onClick={[Function]}
    role="presentation"
  />,
  <div
    className="lab-dialog lab-dialog--message
            "
  >
    <div
      aria-modal="true"
      className="lab-dialog__content lab-dialog__content--message"
      role="dialog"
    >
      <button
        className="lab-dialog__mobile-close-button lab-dialog__mobile-close-button--message"
        onClick={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        type="button"
      >
        <span
          className="lab-icon lab-icon--chevron-down"
        >
          <svg
            aria-hidden={true}
            className="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
            fill="currentColor"
            height={20}
            viewBox="0 0 20 20"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.85 7.65c.2.2.2.5 0 .7l-5.46 5.49a.55.55 0 0 1-.78 0L4.15 8.35a.5.5 0 1 1 .7-.7L10 12.8l5.15-5.16c.2-.2.5-.2.7 0Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </button>
      <div
        className="lab-dialog__header lab-dialog__header--message"
      >
        <button
          className="lab-dialog__close-button"
          onClick={[Function]}
          tabIndex={2}
          type="button"
        >
          <span
            className="lab-icon lab-icon--remove lab-dialog__close-button-icon"
          >
            <svg
              aria-hidden={true}
              className="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height={20}
              viewBox="0 0 20 20"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m4.09 4.22.06-.07a.5.5 0 0 1 .63-.06l.07.06L10 9.29l5.15-5.14a.5.5 0 0 1 .63-.06l.07.06c.18.17.2.44.06.63l-.06.07L10.71 10l5.14 5.15c.18.17.2.44.06.63l-.06.07a.5.5 0 0 1-.63.06l-.07-.06L10 10.71l-5.15 5.14a.5.5 0 0 1-.63.06l-.07-.06a.5.5 0 0 1-.06-.63l.06-.07L9.29 10 4.15 4.85a.5.5 0 0 1-.06-.63l.06-.07-.06.07Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="lab-dialog__icon-wrapper"
      >
        <span
          className="lab-icon lab-icon--star lab-dialog__icon"
        >
          <svg
            aria-hidden={true}
            className="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
            fill="currentColor"
            height={20}
            viewBox="0 0 20 20"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 2a8 8 0 1 1 0 16 8 8 0 0 1 0-16Zm0 1a7 7 0 1 0 0 14 7 7 0 0 0 0-14Zm0 9.5a.75.75 0 1 1 0 1.5.75.75 0 0 1 0-1.5ZM10 6a.5.5 0 0 1 .5.41V11a.5.5 0 0 1-1 .09V6.5c0-.28.22-.5.5-.5Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        className="lab-dialog__title lab-dialog__title--message"
      >
        Sample Message Dialog
      </div>
      <div
        className="lab-dialog__body lab-dialog__body--message"
      >
        This is the main content of a Message Dialog. You may want to put as much content as you want, as long as it's a string.
      </div>
      <div
        className="lab-dialog__footer lab-dialog__footer--message"
      >
        <button
          className="lab-btn lab-btn--outline lab-btn--normal"
          onClick={[Function]}
          tabIndex={3}
          type="button"
        >
          
          Optional button
        </button>
        <button
          className="lab-btn lab-btn--default lab-btn--normal"
          onClick={[Function]}
          tabIndex={1}
          type="button"
        >
          
          Required button
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`MessageDialog rendering with isModal false 1`] = `
<div
  className="lab-dialog lab-dialog--message
             lab-dialog--large"
>
  <div
    aria-modal="true"
    className="lab-dialog__content lab-dialog__content--message"
    role="dialog"
  >
    <button
      className="lab-dialog__mobile-close-button lab-dialog__mobile-close-button--message"
      onClick={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <span
        className="lab-icon lab-icon--chevron-down"
      >
        <svg
          aria-hidden={true}
          className="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
          fill="currentColor"
          height={20}
          viewBox="0 0 20 20"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.85 7.65c.2.2.2.5 0 .7l-5.46 5.49a.55.55 0 0 1-.78 0L4.15 8.35a.5.5 0 1 1 .7-.7L10 12.8l5.15-5.16c.2-.2.5-.2.7 0Z"
            fill="currentColor"
          />
        </svg>
      </span>
    </button>
    <div
      className="lab-dialog__header lab-dialog__header--message"
    >
      <button
        className="lab-dialog__close-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="lab-icon lab-icon--remove lab-dialog__close-button-icon"
        >
          <svg
            aria-hidden={true}
            className="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
            fill="currentColor"
            height={20}
            viewBox="0 0 20 20"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m4.09 4.22.06-.07a.5.5 0 0 1 .63-.06l.07.06L10 9.29l5.15-5.14a.5.5 0 0 1 .63-.06l.07.06c.18.17.2.44.06.63l-.06.07L10.71 10l5.14 5.15c.18.17.2.44.06.63l-.06.07a.5.5 0 0 1-.63.06l-.07-.06L10 10.71l-5.15 5.14a.5.5 0 0 1-.63.06l-.07-.06a.5.5 0 0 1-.06-.63l.06-.07L9.29 10 4.15 4.85a.5.5 0 0 1-.06-.63l.06-.07-.06.07Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </button>
    </div>
    <div
      className="lab-dialog__icon-wrapper"
    >
      <span
        className="lab-icon lab-icon--star lab-dialog__icon"
      >
        <svg
          aria-hidden={true}
          className="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
          fill="currentColor"
          height={20}
          viewBox="0 0 20 20"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 2a8 8 0 1 1 0 16 8 8 0 0 1 0-16Zm0 1a7 7 0 1 0 0 14 7 7 0 0 0 0-14Zm0 9.5a.75.75 0 1 1 0 1.5.75.75 0 0 1 0-1.5ZM10 6a.5.5 0 0 1 .5.41V11a.5.5 0 0 1-1 .09V6.5c0-.28.22-.5.5-.5Z"
            fill="currentColor"
          />
        </svg>
      </span>
    </div>
    <div
      className="lab-dialog__title lab-dialog__title--message"
    >
      Sample Message Dialog
    </div>
    <div
      className="lab-dialog__body lab-dialog__body--message"
    >
      This is the main content of a Message Dialog. You may want to put as much content as you want, as long as it's a string.
    </div>
    <div
      className="lab-dialog__footer lab-dialog__footer--message"
    >
      <button
        className="lab-btn lab-btn--outline lab-btn--normal"
        onClick={[Function]}
        type="button"
      >
        
        Optional button
      </button>
      <button
        className="lab-btn lab-btn--default lab-btn--normal"
        onClick={[Function]}
        type="button"
      >
        
        Required button
      </button>
    </div>
  </div>
</div>
`;

exports[`MessageDialog rendering with isOpen false 1`] = `null`;
